src/openrct2-ui/windows/ShortcutKeys.cpp

```diff
@@ -175,7 +175,7 @@ class ShortcutKeysWindow final : public Window
     std::vector<ShortcutTabDesc> _tabs;
     std::vector<rct_widget> _widgets;
     std::vector<ShortcutStringPair> _list;
-    std::optional<size_t> _highlightedItem;
+    int_fast16_t _highlightedItem;
```

Comment:
This is an odd type to be using. I would have at least thought `int_fast32_t` would be better and the same as what it was using before. I don't see this ever actually being 16-bits unless we are on a 16-bit platform, which is very unlikely.

Reply:
My reasoning is that 16 bits is more than enough to store each index and since memory is not important here, fast types are a good choice. The underlying type is very likely 32 bits or even 64 bits on 64 bit architectures.

Reply:
> My reasoning is that 16 bits is more than enough

I am not sure about that. In `OnScrollMouseOver` you are dealing with screen coords in a scroll view. These can often exceed the limits of 16-bit integers when there are lots of items.